packages/rich-text/src/plugins/CommandPalette/onKeyDown.ts

```diff
@@ -36,6 +36,10 @@ export const createOnKeyDown = (): KeyboardHandler => {
         }
       }
 
+      if (isHotkey('space', event)) {
```

Comment:
I'm not a fan of this

> The only downside to this approach is that the search functionality will not work anymore when you type a space (since the rich text command gets removed).

I think that's a huge downside. Breaking the search functionality may lead to more customer issues. Let's re-think this holistically and find a proper fix. We can start by looking at other similar RT implementations and how they handle that and what are our options.

> to press ESC; But this also blurs the editor and they have to click on the RTE again.

Have you tried re-focusing in the editor in the escape block below? We probably want it to only re-focus when there is a `COMMAND_PROMPT` mark (you can use `isMarkActive` for that) otherwise it may cause accessibility issues.

Reply:
I agree we need to fix this holistically. Ideally we would want a UI for when the command palette mark is active, because now if you search for something and there are no results, everything you type will still be inside the command palette mark.

I will create a ticket for a better fix in the backlog.

For now I implemented the ESC solution. After typing ESC once, it will remove the span, and ESC again will blur the editor. I think we could ship that to unblock the customer